src/renamer.fs

```diff
@@ -4,333 +4,340 @@ open System.Collections.Generic
 open Ast
 open Options.Globals
 
-// Environment for renamer
-// This object is useful to separate the AST walking from the renaming strategy.
-// Maybe we could use a single mutable object, instead of creating envs all the time.
-// TODO: create a real class.
-[<NoComparison; NoEquality>]
-type Env = {
-    // Map from an old variable name to the new one.
-    map: Map<Ident, Ident>
-    // Map from an old function name and function arity to the new name.
-    fct: Map<Ident, Map<int, Ident>>
-    // List of names that are still available.
-    reusable: Ident list
-
-    // Used only for optimizeFrequency.
-    max: int
-
-    // Whether multiple functions can have the same name (but different arity).
-    allowOverloading: bool
-    // Function that decides a name (and returns the new Env).
-    newId: Env -> Ident -> (Env * string)
-    // Function called when we enter a function, optionally updates the Env.
-    enterFunction: Env -> Stmt -> Env
-}
-with
-    static member Create(reusable, allowOverloading, newId, enterFunction) = {
-        map = Map.empty
-        max = 0
-        fct = Map.empty
-        reusable = reusable
-        allowOverloading = allowOverloading
-        newId = newId
-        enterFunction = enterFunction
+module private RenamerImpl =
+
+    // Environment for renamer
+    // This object is useful to separate the AST walking from the renaming strategy.
+    // Maybe we could use a single mutable object, instead of creating envs all the time.
+    // TODO: create a real class.
+    [<NoComparison; NoEquality>]
+    type Env = {
+        // Map from an old variable name to the new one.
+        varRenames: Map<Ident, Ident>
+        // Map from a new function name and function arity to the old name.
+        funRenames: Map<Ident, Map<int, Ident>>
+        // List of names that are still available after having been used before.
```

Comment:
Not exactly: the list should contain names at the beginning. So it's a list of available names.

Reply:
I started calling it availableNames, then I added this comment to keep in mind the reuse aspect, but then I went back to reusableNames...
But maybe it's not that much about availability or reuse, and more about the order, the frequency of use, for compression by packers... Does frequentNames work?